File: Juan/P4/Java/Napakalaki/src/NapakalakiGame/NumericBadConsequence.java

package NapakalakiGame;

import java.util.ArrayList;

/**
 *
 * @author juane
 */
public class NumericBadConsequence extends BadConsequence{
    
    private int nVisibleTreasures; // número de tesoros visibles que si pierden
    private int nHiddenTreasures; // número de tesoros ocultos que se pierden
    private boolean death;

    public NumericBadConsequence(String text, int levels, int nVisible, int nHidden) {
        super(text, levels);
        nVisibleTreasures = nVisible;
        nHiddenTreasures = nHidden;
    }
    
    
    @Override
    public boolean isEmpty(){
        
        if(nHiddenTreasures==0 && nVisibleTreasures==0 && this.death == false ){
            return true;
        }
        else
            return false; 
    }
    
    @Override
    public int geNVisibleTreasures() {
        return nVisibleTreasures;
    }
    
    @Override
    public int getNHiddenTreasures() {
        return nHiddenTreasures;
    }
    
    @Override
    public BadConsequence adjustToFitTreasureList(ArrayList<Treasure> v, ArrayList<Treasure> h) {
        int tamV = v.size();
        int tamH = h.size();
        int nuevoV;
        int nuevoO;

        int nVisibleAux = nVisibleTreasures;
        int nHiddenAux = nHiddenTreasures;

        NumericBadConsequence badConsequence;
        
        //Esto es para los tesoros visibles
        if(nVisibleAux < tamV)
            nuevoV=nVisibleAux;
        else
            nuevoV=tamV;
        //Esto es para los tesoros ocultos
        if(nHiddenAux < tamH)
            nuevoO=nHiddenAux;
        else
            nuevoO=tamH;
        
         badConsequence = new NumericBadConsequence(text, levels, nVisibleAux, nHiddenAux);
         return badConsequence;
        
    } 

    
    
    @Override
    public String toString() {
        return "NumericBadConsequence{" 
                + "BadConsequence = " +  this.text 
                + ", levels = " +  this.levels
                + ", death = " +  this.death
                + "nVisibleTreasures=" + nVisibleTreasures 
                + ", nHiddenTreasures=" + nHiddenTreasures + '}';
    }
    
    
}

File: Juan/P4_FULL/ruby5/Napakalaki/lib/NumericBadConsequence.rb
require_relative 'BadConsequence.rb'
require_relative 'Monster.rb'
require_relative 'Prize.rb'
require_relative 'TreasureKind.rb'
module NapakalakiGame

class NumericBadConsequence < BadConsequence
    
    @@MAXTREASURES = 10
    
    def initialize( text,  levels,  nVisible,  nHidden)
        @nVisibleTreasures=nVisible
        @nHiddenTreasures=nHidden
        super(text,levels)
        
    end
    
    attr_reader :nVisibleTreasures
    attr_reader :nHiddenTreasures
    
    public
    def isEmpty()
        vacio = false
        if @nHiddenTreasures == 0 && @nVisibleTreasures==0  && @death == false 
            vacio = true;
        end
            vacio
    end
    
    def adjustToFitTreasureList(v,h)
       
        tamV = v.size
        tamH = h.size
        
        nVisibleAux = @nVisibleTreasures
        nHiddenAux = @nHiddenTreasures
        
        if(nVisibleAux < tamV)
            nuevoV = nVisibleAux
        else
            nuevoV = tamV
        end
        
        if(nHiddenAux < tamH)
            nuevoH = nHiddenAux
        else
            nuevoH = tamH
        end
        
        badConsequence = NumericBadConsequence.new(@text,@levels,nuevoV,nuevoH)
        return badConsequence
    
    end
    
    def to_s 
        "BadConsequence =  #{@text}  , Levels =  #{@levels}  , nVisibleTreasure = #{@nVisibleTreasures}  nHiddenTreasures =  #{@nHiddenTreasures} , death =  #{@death}"     
    end  
    
    
end
end
